BLSQ/iaso
# IA-1897:  replace SearchFilter component

`SearchFilter`  was only used in OrgUnits, i removed the component and refactored `SearchInput` in bluesquare-components to handle text search errors and use `InputComponent` in `OrgunitsFilters` 

Related JIRA tickets : [IA-1897](https://bluesquare.atlassian.net/browse/IA-1897?atlOrigin=eyJpIjoiYzllMWYyZDI4NWFmNDBhNzkyM2ZkODU5YTU2YWQ0OWQiLCJwIjoiaiJ9)
## Self proofreading checklist


## Changes

- use logic in `SearchFilter` to check for forbiddenCharacters and migrate to `SearchInput` in bluesquare-components 
see [here](https://github.com/BLSQ/bluesquare-components/pull/104)
- pass `blockForbiddenChars` and `onErrorChange` props to `InputComponent` and `OrgunitsFilter` to handle error state and disable/enable `SearchButton`

## How to test

- Go to OrgUnits list
-  type something in `Search` filter and try to add forbidden characters (`['"', '?', '/', '%', '&']`) => the error message appears below the input + the search button should be disabled and the press key on this input should be disabled too

## Print screen / video

https://user-images.githubusercontent.com/25134301/222440672-86360fa2-3855-412b-8dfe-7a1b12301a01.mov

[IA-1897]: https://bluesquare.atlassian.net/browse/IA-1897?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Notes

This [PR](https://github.com/BLSQ/bluesquare-components/pull/104) in bluesquare-components needs to be merged before merging this one
